tools/azsdk-cli/docs/sdk-sample-generation-plan.md

```diff
@@ -0,0 +1,387 @@
+  - Uses helpers and language services to compute correct sample folder and run format/lint after generation.
+  - Uses MicroagentHostService to generate sample code based on user prompts and package context.
+  - Exposes an MCP method for agent-triggered generation.
+- Response model for structured output and JSON mode support.
+- Registration in CLI tool list; no service registration changes required beyond using existing services.
+
+## Files to add/edit
+
+- Add tool implementation
+  - New file: `Azure.Sdk.Tools.Cli/Tools/Generators/SampleGeneratorTool.cs`
+    - Namespace: `Azure.Sdk.Tools.Cli.Tools.Generators`
+    - Inherit from `MCPTool` ([contract](../Azure.Sdk.Tools.Cli.Contract/MCPTool.cs))
+    - CommandHierarchy = `[SharedCommandGroups.Generators]` ([SharedCommandGroups.cs](../Azure.Sdk.Tools.Cli/Commands/SharedCommandGroups.cs))
+    - Follow patterns from [ReadMeGeneratorTool](../Azure.Sdk.Tools.Cli/Tools/Package/ReadMeGeneratorTool.cs)
+
+- Add Docker service for container operations
+  - New file: `Azure.Sdk.Tools.Cli/Services/IDockerService.cs`
+    - General-purpose interface for Docker container management and operations
+  - New file: `Azure.Sdk.Tools.Cli/Services/DockerService.cs`
+    - Implementation using Docker CLI via `IProcessHelper`
+    - Methods: `CreateContainerAsync`, `RunCommandInContainerAsync`, `CopyToContainerAsync`, `CopyFromContainerAsync`, `RemoveContainerAsync`, etc.
+
+- Register tool with CLI
+  - Edit: `Azure.Sdk.Tools.Cli/Commands/SharedOptions.cs`
+    - Add `typeof(SampleGeneratorTool)` to `ToolsList` ([link](../Azure.Sdk.Tools.Cli/Commands/SharedOptions.cs))
+
+- Response type
+  - New file: `Azure.Sdk.Tools.Cli/Models/Responses/SamplesGenerationResponse.cs`
+    - Inherit from `Response`
+    - Include: `FilesWritten`, `Warnings`, `Messages`, `DryRun`, `Language`, `OutputRoot`
+
+- Tests
+  - New file: `Azure.Sdk.Tools.Cli.Tests/Tools/Generators/SampleGeneratorToolTests.cs`
+    - Mirror test structure from [ReadMeGeneratorToolTests](../Azure.Sdk.Tools.Cli.Tests/Tools/Generators/ReadMeGeneratorToolTests.cs)
+    - Unit test the CLI handler argument parsing and a mocked MCP flow
+
+- Docs
+  - Update `Azure.Sdk.Tools.Cli/README.md` usage section to include examples for the new command
+
+## Command contract (CLI)
+
+- Command path: `azsdk generators samples`
+- Options
+  - `--package-path` (required): Absolute or repo-relative path to the package folder (under a language repo `sdk/...`)
+  - `--prompt` (required): User description of the sample(s) to generate. Can describe a single scenario (e.g., "Create a sample that uploads a file to blob storage") or multiple scenarios (e.g., "Create samples for: 1) uploading a file, 2) downloading a file, 3) listing blobs in a container"). When multiple scenarios are described, separate sample files will be generated for each scenario.
```

Comment:
Do scenarios need to be explicitly numbered or can we prompt with something more open ended such as - "create samples for any API scenarios that do not already have samples"